src/vcpkg/binarycaching.cpp

```diff
@@ -211,18 +213,443 @@ namespace
+                        write_settings_file();
+                    }
+                }
+            }
+            return maybe_settings;
+        }
+
+        WriteFilePointer get_own_sync_file(const Path& sync_root_dir)
+        {
+            while (true)
+            {
+                Path path = sync_root_dir / fmt::format("{}", rand());
+                std::error_code ec;
+                WriteFilePointer wp(path, Append::NO, Overwrite::NO, ec);
+                if (!ec)
+                {
+                    return wp;
+                }
+            }
+        }
+
         size_t push_success(const BinaryPackageWriteInfo& request, MessageSink& msg_sink) override
         {
             const auto& zip_path = request.zip_path.value_or_exit(VCPKG_LINE_INFO);
             const auto archive_subpath = files_archive_subpath(request.package_abi);
 
+            size_t index = 0;
+            uint64_t file_size = m_fs.file_size(zip_path, IgnoreErrors{});
             size_t count_stored = 0;
             for (const auto& archives_root_dir : m_dirs)
             {
+                if (index >= file_cache_data.size())
+                {
+                    auto& cache = file_cache_data.emplace_back(
+                        FileCacheData{get_folder_settings(archives_root_dir, msg_sink).value_or({})});
+                    cache.sync_root_dir = archives_root_dir / "sync";
+                    m_fs.create_directories(cache.sync_root_dir, VCPKG_LINE_INFO);
+                    cache.own_sync_file = get_own_sync_file(cache.sync_root_dir);
+                    if (cache.folder_settings.delete_policy != FolderSettings::DeletePolicy::None)
+                    {
+                        std::unordered_map<std::string, uint64_t> file_sizes;
+                        cache.get_sync_updates(
+                            m_fs,
+                            [&](auto id, auto size) {
+                                auto size_as_int = Strings::strto<uint64_t>(size).value_or_exit(VCPKG_LINE_INFO);
+                                file_sizes.emplace(id.to_string(), size_as_int);
+                            },
+                            true);
+                        auto& settings = cache.folder_settings;
+                        auto settings_path = archives_root_dir / "settings.json";
+                        for (auto& path : m_fs.get_regular_files_recursive(archives_root_dir, IgnoreErrors{}))
+                        {
+                            if (path.filename() == ".DS_Store" || path == settings_path)
+                            {
+                                continue;
+                            }
+                            if (path.parent_path() == cache.sync_root_dir)
+                            {
+                                continue;
+                            }
+                            auto time = settings.last_time(m_fs, path, IgnoreErrors{});
+                            const auto size = std::invoke([&]() {
```

Comment:
Why are you immediately invoking lambdas through `std::invoke`? The area under this code and the throughput of doing this makes my compiler cry

Reply:
The make it more obvious, but I can also make the compiler happy 😆